packages/web-components/src/menu-list/menu-list.spec.ts

```diff
@@ -521,4 +523,67 @@ test.describe('Menu', () => {
       await expect(item).toHaveAttribute('data-indent', '2');
     }
   });
+
+  test.describe('`change` event', () => {
+    test('should emit `change` event when `checked` property changed', async ({ page }) => {
+      const element = page.locator('fluent-menu-list');
+      const menuItems = element.locator('fluent-menu-item');
+
+      await page.setContent(/* html */ `
+      <fluent-menu-list>
+        <fluent-menu-item role="menuitemradio">Menu Item 1</fluent-menu-item>
+        <fluent-menu-item>Menu item 2</fluent-menu-item>
+        <fluent-menu-item>Menu item 3</fluent-menu-item>
+        <fluent-menu-item>Menu item 4</fluent-menu-item>
+      </fluent-menu-list>
+      `);
+
+      const [wasChanged] = await Promise.all([
+        menuItems.nth(0).evaluate(
+          node => new Promise(resolve => node.addEventListener('change', () => resolve(true), { once: true })),
+        ),
+        menuItems.nth(0).evaluate((node: MenuItem) => {
+          node.checked = true;
+        }),
+      ]);
+
+      expect(wasChanged).toEqual(true);
```

Comment:
Here's a cleaner way to handle the async calls:

```suggestion
      const wasChanged = menuItems.nth(0).evaluate(node => {
        return new Promise(resolve => {
          node.addEventListener('change', () => {
            resolve(true);
          });
        });
      });

      await menuItems.nth(0).evaluate((node: MenuItem) => {
        node.checked = true;
      });

      await expect(wasChanged).resolves.toEqual(true);
```